### PR TITLE
[code] Pass GITPOD_CODE_HOST when starting vscode server

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Related https://github.com/gitpod-io/openvscode-server/commit/8bcdbe5e6e097191649984c8329af15a3e6b3066

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at a526592</samp>

Switched the code-server component to use the open-vsx proxy for extensions and refactored the codehelper package to support different domains. Simplified the code-server Dockerfile and updated the workspace configuration.

</details>

## Related Issue(s)

Fixes #

## How to test

Check both vscode stable and insiders works:
- Extension gallery
- Settings sync
- Webviews
- Web extensions (vim)

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - jp-tempora4ddbe88d45</li>
	<li><b>🔗 URL</b> - <a href="https://jp-tempora4ddbe88d45.preview.gitpod-dev.com/workspaces" target="_blank">jp-tempora4ddbe88d45.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - jp-temporary-seahorse-gha.17069</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-jp-tempora4ddbe88d45%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by jeanp413** [comment]
Seems there's a bug in our open-vsx proxy:
- https://open-vsx.gitpod.io/vscode/unpkg/vscodevim/vim/1.26.0/extension is not working, it should return same response as https://open-vsx.org/vscode/unpkg/vscodevim/vim/1.26.0/extension

File: components/ide/code/codehelper/main.go [hunk 1]
```diff
@@ -61,14 +61,21 @@ func main() {
 	}
 	phaseDone()
 
+	url, err := url.Parse(wsInfo.GitpodHost)
```
Comment (akosyakov) [inline components/ide/code/codehelper/main.go:64]: We have only codehelper which should work with all images. Let's keep backward compatible mode based on version from product.json.

We will need to keep old rules in installer as well.

We can only remove them when all Dedicated cells and Cloud are migrated to new images.

File: install/installer/pkg/components/blobserve/configmap.go [hunk 2]
```diff
@@ -49,21 +49,13 @@ func configmap(ctx *common.RenderContext) ([]runtime.Object, error) {
...
-						Replacement: extensionsGalleryItemUrl,
-						Path:        "/ide/out/vs/workbench/workbench.web.api.js",
 					}, {
 						Search:      "{{extensionsGalleryItemUrl}}",
 						Replacement: extensionsGalleryItemUrl,
 						Path:        "/ide/out/vs/workbench/workbench.web.main.js",
-					}, {
-						Search:      "{{trustedDomain}}",
-						Replacement: trustedDomain,
-						Path:        "/ide/out/vs/workbench/workbench.web.api.js",
```
Comment (jeanp413) [inline install/installer/pkg/components/blobserve/configmap.go:66]: `workbench.web.api.js` file does not exist a long time ago so it's fine to delete these rules

File: install/installer/pkg/components/blobserve/configmap.go [hunk 3]
```diff
@@ -76,20 +68,14 @@ func configmap(ctx *common.RenderContext) ([]runtime.Object, error) {
...
-						Replacement: ".",
-					}, {
-						Search:      "value.startsWith(window.location.origin)",
-						Replacement: "value.startsWith(window.location.origin) || value.startsWith('${ide}')",
 					}, {
 						Search:      "./out",
 						Replacement: "${ide}/out",
-					}, {
-						Search:      "./node_modules",
-						Replacement: "${ide}/node_modules",
```
Comment (jeanp413) [inline install/installer/pkg/components/blobserve/configmap.go:92]: These search values do not exist in `index.html` a long time ago so removing them it's fine

File: components/ide/code/codehelper/main.go (hunk 1 above)
Comment (jeanp413) [reply components/ide/code/codehelper/main.go:64]: This is just :lipstick: as `GitpodHost` was parsed twice in `prepareServerMain` and `prepareWebWorkbenchMain`, the functions will still be invoked correctly for all versions, `GITPOD_CODE_HOST` env var will be ignored for old versions

**Comment by jeanp413** [comment]
`resourceUrlTemplate: 'https://open-vsx.org/vscode/unpkg/{publisher}/{name}/{version}/{path}'` will be harcoded in gp-code until open-vsx proxy is fixed

**Review by akosyakov — APPROVED** [review]
i did not try

**Comment by jeanp413** [comment]
Tested stable and insiders, both work :tada:

**Pull request merged**